crates/driver/src/tests/setup/mod.rs

```diff
@@ -803,6 +804,7 @@ impl Setup {
             let quote = blockchain.quote(&order).await;
             quotes.push(quote);
         }
+        let (tx, rx) = std::sync::mpsc::channel();
```

Comment:
I believe we are using `tokio`, if our runtime is `tokio` and we use `std::sync`, we can potentially block the thread, it is dangerous. `std::sync`, especially `recv()`, uses threading blocking operations, whereas `tokio` runtime uses a poll of tasks which don't necessarily run in separate threads. See https://ryhl.io/blog/async-what-is-blocking/

Tokio has its own `mpsc`, or `flume` (`tokio` compatible and high performance).

Reply:
Generally I agree, but here we are in a test case, where a sync API is preferable because I want to enforce that we check all notifications when we drop each `Test`. We could do this with tokio spawn blocking but then I don't think we would win anything.

Also, we are using `try_recv` which is non blocking.

Reply:
oh, it is a test case, my bad, I oversaw that, sorry. I fully agree with you!